### PR TITLE
feat: [UIE-8131] - IAM RBAC: Routes, Menu, Feature Flag

### PR DESCRIPTION
**Pull request by aaleksee-akamai**
## Description 📝

IAM RBAC - all routes, menu, feature flag

## Changes  🔄
- new menu item
- new feature flag
- routes for iam

## Target release date 🗓️

Dec 10, 2024 (dev)

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| image
 | image
 |

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

---

### CODE REVIEW

File: packages/manager/src/features/IAM/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const IdentityAccessManagement: React.FC<CombinedProps> = (props) => {
+  const path = props.match.path;
+
+  return (
+    <React.Suspense fallback={<SuspenseLoader />}>
+      <ProductInformationBanner bannerLocation="Identity and Access Management" />
+      <Switch>
+        <Route component={UserDetails} path={`${path}/users/:username/`} />
+        <Redirect exact from={path} to={`${path}/users`} />
+        <Route component={IAMLanding} path={`${path}`} />
```
Comment (bnussman-akamai) [inline packages/manager/src/features/IAM/index.tsx:28]:
```suggestion
        <Route component={IAMLanding} path={path} />
```

File: packages/manager/src/features/IAM/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+const UserDetails = React.lazy(() =>
+  import('./Users/UserDetailsLanding').then((module) => ({
+    default: module.UserDetailsLanding,
+  }))
+);
+
+type CombinedProps = RouteComponentProps;
+
+export const IdentityAccessManagement: React.FC<CombinedProps> = (props) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/IAM/index.tsx:19]: We can express this in a cleaner way. We aren't combining any types to we don't need a type called `CombinedProps`. We also don't need to use `React.FC`. 

```suggestion
export const IdentityAccessManagement = (props: RouteComponentProps) => {
```

File: packages/manager/src/features/IAM/IAMLanding.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              <Roles />
+            </SafeTabPanel>
+          </TabPanels>
+        </React.Suspense>
+      </Tabs>
+    </>
+  );
+});
+
+export default IdentityAccessManagementLanding;
```
Comment (bnussman-akamai) [inline packages/manager/src/features/IAM/IAMLanding.tsx:86]: For consistency, can we just remove this default export  and export `IdentityAccessManagementLanding`

**Review by bnussman-akamai — COMMENTED** [review]
Looks good. A small number of things need to be cleaned up

File: packages/manager/src/features/IAM/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      <Switch>
+        <Route component={UserDetails} path={`${path}/users/:username/`} />
+        <Redirect exact from={path} to={`${path}/users`} />
+        <Route component={IAMLanding} path={`${path}`} />
+      </Switch>
+    </React.Suspense>
+  );
+};
+
+export default IdentityAccessManagement;
```
[the remark this answers is not in the document]
Comment (aaleksee-akamai) [reply packages/manager/src/features/IAM/index.tsx]: sure, done

File: packages/manager/src/features/IAM/index.tsx (hunk 1 above)
Comment (aaleksee-akamai) [reply packages/manager/src/features/IAM/index.tsx]: done

File: packages/manager/src/features/IAM/index.tsx (hunk 2 above)
Comment (aaleksee-akamai) [reply packages/manager/src/features/IAM/index.tsx]: done

**Comment by aaleksee-akamai** [comment]
@bnussman-akamai , thanks, I've updated this PR

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**